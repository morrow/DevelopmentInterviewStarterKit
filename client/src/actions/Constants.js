export const RECEIVE_ME = 'RECEIVE_ME';
export const RECEIVE_PEOPLE = 'RECEIVE_PEOPLE';