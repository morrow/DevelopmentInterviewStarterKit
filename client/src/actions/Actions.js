import * as actions from './Constants'

export function receiveMe(me) {
  return {
    type: actions.RECEIVE_ME,
    me
  }
}

export function receivePeople(people) {
  return {
    type: actions.RECEIVE_PEOPLE,
    people
  }
}

export function showEmailCharacterFrequency(){
  return {
    type: actions.SHOW_EMAIL_CHARACTER_FREQUENCY,
  }
}

export function hideEmailCharacterFrequency(){
  return {
    type: actions.HIDE_EMAIL_CHARACTER_FREQUENCY,
  }
}

export function toggleEmailCharacterFrequency(){
  return {
    type: actions.TOGGLE_EMAIL_CHARACTER_FREQUENCY,
  }
}

export function showPossibleDuplicates(){
  return {
    type: actions.SHOW_POSSIBLE_DUPLICATES,
  }
}

export function hidePossibleDuplicates(){
  return {
    type: actions.HIDE_POSSIBLE_DUPLICATES,
  }
}

export function togglePossibleDuplicates(){
  return {
    type: actions.TOGGLE_POSSIBLE_DUPLICATES,
  }
}