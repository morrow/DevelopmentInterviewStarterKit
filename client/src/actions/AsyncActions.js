import * as actions from './Actions'

export const creds = { credentials: 'same-origin' };

export function fetchMe() {
  return dispatch => {
    return fetch('/api/me.json', creds)
      .then(response => response.json())
      .then(me => dispatch(actions.receiveMe(me)));
  }
}

export function fetchPeople() {
  return dispatch => {
    return fetch('/api/people.json', creds)
      .then(response => response.json())
      .then(people => dispatch(actions.receivePeople(people)))
  }
}