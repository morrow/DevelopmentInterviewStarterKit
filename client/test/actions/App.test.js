import configureStore from 'redux-mock-store'
import configureMockStore from 'redux-mock-store'

import * as actions from '../../src/actions/Actions'
import * as constants from '../../src/actions/Constants'

describe('app actions', () => {

  let store

  beforeEach(() => {
    const mockStore = configureStore()
    store = mockStore({
      app:[]
    })
    store.clearActions();
  });

  describe('showEmailCharacterFrequency', () => {

    test('Dispatches the showEmailCharacterFrequency action', () => {
      store.dispatch(actions.showEmailCharacterFrequency());
      expect(store.getActions()).toMatchSnapshot();
    });

  });

  describe('hideEmailCharacterFrequency', () => {

    test('Dispatches the hideEmailCharacterFrequency action', () => {
      store.dispatch(actions.hideEmailCharacterFrequency());
      expect(store.getActions()).toMatchSnapshot();
    });

  });

  describe('toggleEmailCharacterFrequency', () => {

    test('Dispatches the toggleEmailCharacterFrequency action', () => {
      store.dispatch(actions.toggleEmailCharacterFrequency());
      expect(store.getActions()).toMatchSnapshot();
    });

  });


});