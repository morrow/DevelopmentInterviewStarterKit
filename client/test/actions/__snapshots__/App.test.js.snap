// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app actions hideEmailCharacterFrequency Dispatches the hideEmailCharacterFrequency action 1`] = `
Array [
  Object {
    "type": "HIDE_EMAIL_CHARACTER_FREQUENCY",
  },
]
`;

exports[`app actions showEmailCharacterFrequency Dispatches the showEmailCharacterFrequency action 1`] = `
Array [
  Object {
    "type": "SHOW_EMAIL_CHARACTER_FREQUENCY",
  },
]
`;

exports[`app actions toggleEmailCharacterFrequency Dispatches the toggleEmailCharacterFrequency action 1`] = `
Array [
  Object {
    "type": "TOGGLE_EMAIL_CHARACTER_FREQUENCY",
  },
]
`;
